app/models/provisioning_template.rb

```diff
@@ -97,7 +97,7 @@ def clone
 
   def self.find_template(opts = {})
     raise ::Foreman::Exception.new(N_("Must provide template kind")) unless opts[:kind]
-    raise ::Foreman::Exception.new(N_("Must provide an operating system")) unless opts[:operatingsystem_id]
+    raise ::Foreman::Exception.new(N_("Host is not associated with an operating system")) unless opts[:operatingsystem_id]
```

Comment:
This method feels like an internal API and it looks like this is returning a user error where it should really return an API error; in other words, solving it at the wrong layer. Here we don't have anything like host, just a set of options. Perhaps the register controller should first ensure an OS is associated before calling this. If not, present the user with the correct error.

The alternative is to do so in the HostExtension. There you can also provide context about which host it is, rather than some unknown host.